test/unittest/dds/participant/ParticipantTests.cpp

```diff
@@ -799,30 +894,44 @@ TEST(ParticipantTests, RemoteServersListConfiguration)
     file << "{\"ROS_DISCOVERY_SERVER\": \"172.17.0.5:4321;192.168.1.133:64863\"}";
```

Comment:
Shouldn't this fail since we already defined a Discovery Server via environment variable earlier and it is not included here?

Reply:
That is just the issue this test intends to check: what happens when two servers with the same guidPrefix are being added, one through the environment variable and another through QoS. It seems that both are included though only one of them is pinged (as the test comment explains). Also, the attributes only holds the one set by QoS (which is not the one being pinged as far as I know checking with a wireshark capture) and it is against this server that the feature checks that no server has been removed. That is the reason why this test is called inconsistent. The main goal was checking that Fast DDS does not have a core and that any mishandle of the API is safe (even though the behavior can be undefined). That is the reason why in [our documentation](https://fast-dds.docs.eprosima.com/en/latest/fastdds/discovery/discovery_server.html#modifying-remote-servers-list-at-run-time) we have included a note stating that it is advised to use either the API or the environment variable but not both at the same time.